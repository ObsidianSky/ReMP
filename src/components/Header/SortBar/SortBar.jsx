import React, { Component } from 'react';
import Bar from '../../common/Bar/Bar';
import SearchFilter from '../SearchFilter/SearchFilter';

import { connect } from 'react-redux';
import { sortMovies } from '../../../actions';

class BottomBar extends Component {
    render() {
        if(this.props.moviesAmount) {
            return (
                <Bar title={`${this.props.moviesAmount} movies found`}>
                    <SearchFilter filters={this.props.sortTypes}
                                  type="sort"
                                  activeFilter={this.props.selectedSortType}
                                  onSelect={this.props.sortMovies}
                                  title="Sort by"/>
                </Bar>
            );
        } else {
            return <Bar/>;
        }
    }
}

const mapStateToProps = ({ movies }) => ({
    moviesAmount: movies.items.length,
    sortTypes: movies.sortTypes,
    selectedSortType: movies.selectedSortType
});

export default connect(mapStateToProps, { sortMovies } )(BottomBar);