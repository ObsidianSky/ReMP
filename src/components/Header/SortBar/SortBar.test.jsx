import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { SortBar } from './SortBar';

describe('SortBar', () => {
    let props;

    beforeEach(() => {
        props = {
            moviesAmount: 2,
            sortTypes: [
                { label: 'release date', value: 'year' },
                { label: 'rating', value: 'rating' }
            ],
            selectedSortType: 'selectedSortType',
            sortMovies: () => {},
        }
    });

    describe('when we have movies', () => {
        test('should renders correctly', () => {
            const wrapper = shallow(<SortBar { ...props } />);

            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('when we do NOT have movies', () => {
        beforeEach(() => {
            props.moviesAmount = 0;
        });

        test('should renders correctly', () => {
            const wrapper = shallow(<SortBar { ...props } />);

            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });
});

