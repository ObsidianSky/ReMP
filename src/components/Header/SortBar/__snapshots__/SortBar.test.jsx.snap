// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBar when we do NOT have movies should renders correctly 1`] = `
<Bar
  title=""
/>
`;

exports[`SortBar when we have movies should renders correctly 1`] = `
<Bar
  title="2 movies found"
>
  <RadioButtonGroup
    activeOption="selectedSortType"
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "release date",
          "value": "year",
        },
        Object {
          "label": "rating",
          "value": "rating",
        },
      ]
    }
    title="Sort by"
    type="sort"
  />
</Bar>
`;
